From: Openkylin Developers <packaging@lists.openkylin.top>
Date: Thu, 8 Sep 2022 02:10:22 +0000
Subject: Remove dummy `ucontext` declaration

This type is defined somewhere in the system, no need to make a dummy one.
---
 libbacktrace/include/backtrace/Backtrace.h | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/libbacktrace/include/backtrace/Backtrace.h b/libbacktrace/include/backtrace/Backtrace.h
index b919e81..5dbc3c8 100644
--- a/libbacktrace/include/backtrace/Backtrace.h
+++ b/libbacktrace/include/backtrace/Backtrace.h
@@ -26,6 +26,8 @@
 #include <backtrace/backtrace_constants.h>
 #include <backtrace/BacktraceMap.h>
 
+#include <sys/ucontext.h>
+
 #if __LP64__
 #define PRIPTR "016" PRIxPTR
 typedef uint64_t word_t;
@@ -66,9 +68,6 @@ struct backtrace_frame_data_t {
 #if defined(__APPLE__)
 struct __darwin_ucontext;
 typedef __darwin_ucontext ucontext_t;
-#else
-struct ucontext;
-typedef ucontext ucontext_t;
 #endif
 
 struct backtrace_stackinfo_t {
